types/src/transaction/metadata.rs

```diff
@@ -128,3 +129,16 @@ pub struct CoinTradeMetadataV0 {
     /// A list of trade_ids this transaction wants to settle
     pub trade_ids: Vec<String>,
 }
+
+/// List of supported transaction metadata format versions for transactions for payments
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub enum PaymentMetadata {
+    PaymentMetadataVersion0(PaymentMetadataV0),
+}
+
+/// Transaction metadata format for transactions for payments
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct PaymentMetadataV0 {
+    /// Reference ID needed for off-chain reference ID exchange.
+    reference_id: Option<String>,
```

Comment:
IIRC Reference Id's shouldn't be a string but a fixed length of bytes.

Reply:
https://github.com/diem/dip/blob/0ef561f3ed8bf079def20a656d6c85d1c75b3391/dips/dip-10.md#on-chain-transaction-settlement

Specifically:

```suggestion
    reference_id: [u8; 16],
```

Reply:
Btw any idea why all the fields are Option<x> vs just x here? Like the travelrulemetadata one, it says Option<string> for reference ID

Reply:
No I don't know, most of these things were added with little thought unfortunately. We should try to be very strict and concrete with newly added fields.